gammapy/estimators/utils.py

```diff
@@ -536,6 +542,79 @@ def compute_lightcurve_doublingtime(lightcurve, flux_quantity="flux"):
     return table
 
 
+def compute_lightcurve_discrete_correlation(
+    lightcurve1, lightcurve2=None, flux_quantity="flux", tau=1 * u.h
+):
+    r"""Compute the discrete correlation function for two lightcurves, or the discrete autocorrelation if only one lightcurve is provided.
+    NaN values will be ignored in the computation in order to account for possible gaps in the data.
+
+    Internally calls the `~gammapy.stats.discrete_correlation` function
+
+    Parameters
+    ----------
+    lightcurve1 : `~gammapy.estimators.FluxPoints`
+        The first lightcurve object.
+    lightcurve2 : `~gammapy.estimators.FluxPoints`, optional
+        The second lightcurve object. If not provided, the autocorrelation for the first lightcurve will be computed. Default is None.
+    flux_quantity : str
+        Flux quantity to use for calculation. Should be 'dnde', 'flux', 'e2dnde' or 'eflux'. Default is 'flux'.
+    tau : `~astropy.units.Quantity`
+        Size of the bins to compute the discrete correlation. Default is 1 hour.
```

Comment:
I did not mean that tau shoudl always be obtained automatically. I was thinking of replacing the default 1 hr with something more aligned with the time bins of the light curves

Reply:
Thinking more about it, a static default is not great, as you point out. A default in general is hard to fix and I guess would be overridden most of the times. to keep it very simple, it could be something like (length of the time axis)/(N/2) where N is the number of current bins.